src/c/tb_client/context.zig

```diff
@@ -106,17 +108,23 @@ pub fn ContextType(
             context.client_id = std.crypto.random.int(u128);
             log.debug("{}: init: initializing", .{context.client_id});
 
+            const packets_count_max = 4096;
```

Comment:
FWIW, the client could be the one to allocate and provide the packets, but having `tb_client` do the allocation makes language client implementations just a bit easier.

Reply:
Since the packets/concurrency count is application-supplied and the maximum packets is arbitrary, why have a maximum at all?

Reply:
> Since the packets/concurrency count is application-supplied and the maximum packets is arbitrary, why have a maximum at all?

Allowing the application to set some huge/random number can have bad effects (probably panic due to out of memory). I think it is important to define an upper limit or at least change the data type from `u32` to `u16` (could still be huge, but more realistic).

> the client could be the one to allocate and provide the packets, but having tb_client do the allocation makes language client implementations just a bit easier.

We can leave it to each client implementation to check `if (packets_count > 0)` so they can choose how to allocate.